### ISSUE TITLE
Result of running MEND on VQA dataset using minigpt4 model

### ISSUE THREAD

**Issue by yaohui120** [zjunlp/EasyEdit#177]
I changed item['locality_prompt'] and item['locality_ground_truth'] in easyeditor/dataset/vqa.py, using the questions with only one answer in NQ-open.train.jsonl
```
item['locality_prompt'] = self.nq_qa[i%nq_qa_len]['question'] # record['loc']
item['locality_ground_truth'] = self.nq_qa[i%nq_qa_len]['answer'][0] # record['loc_ans']
```
But the results are so different
```
rewrite_acc: 0.04476018114652962
rephrase_acc: 0.04088696036096057
rephrase_image_acc: 0.0508587181062418
locality_acc: 1.0
locality_image_acc: 1.0
```

**Comment by tbozhong** [comment]
Thanks for your attention to our work!

![image](https://github.com/zjunlp/EasyEdit/assets/50789998/a8d50e86-8105-45dc-ac8e-a43c6dfadb20)
I recently tested the code and did not encounter the issue you mentioned. 
Could you please verify that you have properly loaded the checkpoint? Additionally, providing more information would be helpful for me to reproduce the problem and assist you further.

**Comment by yaohui120** [comment]
Thank you! I will check the code, maybe I used the wrong function to evaluate.

**Comment by yaohui120** [comment]
Could you tell me the approximate value of the loss as the step changes? I would like to see if the training process is correct. Thank you!

**Comment by tbozhong** [comment]
![image](https://github.com/zjunlp/EasyEdit/assets/50789998/5840875b-6b8a-4598-ae5e-bfcfe877a470)

You can refer to the provided log. At around 5000 steps, the loss should decrease to approximately 1.0.

Thank you! I have made the updates.

**Comment by yaohui120** [comment]
Could you please help me check the pretrained-model I downloaded for minigpt4? Thank you very much!
tokenizer_name: hugging_cache/Vicuna from 
  - https://huggingface.co/lmsys/vicuna-13b-v1.5/tree/main
qformer_checkpoint: hugging_cache/blip2_pretrained_flant5xxl.pth from
  - https://storage.googleapis.com/sfr-vision-language-research/LAVIS/models/BLIP2/blip2_pretrained_flant5xxl.pth
qformer_name_or_path: hugging_cache/bert-base-uncased from
  - https://huggingface.co/bert-base-uncased/tree/main
state_dict_file: hugging_cache/eva_vit_g.pth from
  - https://storage.googleapis.com/sfr-vision-language-research/LAVIS/models/BLIP2/eva_vit_g.pth
pretrained_ckpt: hugging_cache/pretrained_minigpt4.pth from
  - https://github.com/Vision-CAIR/MiniGPT-4?tab=readme-ov-file( MiniGPT-4 (Vicuna 13B) )

**Comment by tbozhong** [comment]
If you're attempting to edit Vicuna-13B, you should try to select layers 38 and 39 (the final two layers) for your edits.

**Comment by yaohui120** [comment]
When I tried to edit Vicuna-7b using VQA, it happens l_edit = nan several times. But it doesn't happen when I tried to edit Vicuna-13b. So did you ever encounter this problem?

**Comment by tbozhong** [comment]
You can debug in `edit_step` to find the issue, which possibly caused by label length mismatch or model precision (fixed by choosing a suitable `eps`).

**Comment by yaohui120** [comment]
The problem about nan loss can be soluted by setting torch.dtype=torch.bfloat16 when loading self.llama_model.[Here](https://github.com/facebookresearch/llama-recipes/issues/146).
But a new problem is info_dict['grad'] in function train_step in MultimodalTrainer.py is always 0. The l_total_edit is normal and requires_grad is on. After execute safe_backward function, the params' grad of self.model is None.

**Comment by tbozhong** [comment]
Hi, do you have any further questions?

**Comment by yaohui120** [comment]
No, thank you!
